File: unit6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = 0
length = len(text)
for i in range(length):
    if text[i].islower():
        letters = letters + 1
    if text[i].isupper():
        letters = letters + 1

sentences = 0
for i in range(length):
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences = sentences + 1

words = 1
for i in range(length):
    if text[i] == " ":
        words = words + 1

L = 0
L = float(letters) / float(words) * 100

S = 0
S = float(sentences) / float(words) * 100

index = 0
index = 0.0588 * L - 0.296 * S - 15.8

grade = round(index)
if grade >= 1 and grade <= 16:
    print(f"Grade {grade}")
if grade < 1:
    print(f"Before Grade 1")
if grade > 16:
    print(f"Grade 16+")

File: unit2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    int letters = 0;
    int length = strlen(text);
    for (int i = 0; i < length; i = i + 1)
    {
        if (islower(text[i]))
        {
            letters = letters + 1;
        }
        if (isupper(text[i]))
        {
            letters = letters + 1;
        }
    }
    // printf("Letters = %i\n", letters);

    int sentences = 0;
    for (int i = 0; i < length; i = i + 1)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences = sentences + 1;
        }
    }
    // printf("Sentences = %i\n", sentences);

    int words = 1;
    for (int i = 0; i < length; i = i + 1)
    {
        if (text[i] == ' ')
        {
            words = words + 1;
        }
    }
    // printf("Words = %i\n", words);

    float L = 0;
    // L is the average number of letters per 100 words in the text
    L = (float) letters / (float) words * 100;
    // printf("letters per 100 words = %.2f\n", L);

    float S = 0;
    // S is the average number of sentences per 100 words in the text
    S = (float) sentences / (float) words * 100;
    // printf("sentences per 100 words = %.2f\n", S);

    float index = 0;
    index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);
    if (grade >= 1 && grade <= 16)
    {
        printf("Grade %i\n", grade);
    }
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    if (grade > 16)
    {
        printf("Grade 16+\n");
    }
}